src/cloudant/indexes.py

```diff
@@ -0,0 +1,271 @@
+    be used by the database convenience methods
+    :func:`~cloudant.database.CloudantDatabase.create_index`,
+    :func:`~cloudant.database.CloudantDatabase.delete_index`, and
+    :func:`~cloudant.database.CloudantDatabase.get_all_indexes`.  It is
+    recommended that you use those methods to manage an index rather than
+    directly interfacing with SearchIndex objects.
+
+    :param CloudantDatabase database: A Cloudant database instance used by the
+        SearchIndex.
+    :param str design_document_id: Optional identifier of the design document.
+    :param str name: Optional name of the index.
+    :param **kwargs: Options used to construct the index definition for the
+        purposes of index creation.  For more details on valid options See
+        :func:`~cloudant.database.CloudantDatabase.create_index`.
+    """
+    def __init__(self, database, design_document_id=None, name=None, **kwargs):
+        super(SearchIndex, self).__init__(
+            database,
+            design_document_id,
+            name,
+            **kwargs
+        )
+        self._type = 'text'
+
+    def _def_check(self):
+        """
+        Checks that the definition provided contains only valid arguments for a
+        text index.
+        """
+        valid_args = {
+            'fields': list,
+            'default_field': dict,
+            'selector': dict
+        }
+        if self._def != {}:
+            for key, val in self._def.iteritems():
+                if key not in valid_args.keys():
+                    msg = 'Invalid argument: {0}'.format(key)
+                    raise CloudantArgumentError(msg)
+                if not isinstance(val, valid_args[key]):
+                    msg = (
+                        'Argument {0} is not an instance of expected type: {1}'
+                    ).format(key, valid_args[key])
+                    raise CloudantArgumentError(msg)
+
+class SpecialIndex(Index):
+    """
+    Provides an interface for viewing the "special" primary index of a database.
+    Primarily meant to be used by the database convenience method
+    :func:`~cloudant.database.CloudantDatabase.get_all_indexes`.  It is
+    recommended that you use that method to view the special index rather than
+    directly interfacing with the SpecialIndex object.
+
+    :param CloudantDatabase database: A Cloudant database instance used by the
+        SpecialIndex.
+    :param **kwargs: Only an ascending order _id field parameter is allowed as
+        an argument for a SpecialIndex.
+        For example ``fields=[{'_id': 'asc'}]``.
+    """
+    def __init__(self, database, **kwargs):
+        if kwargs.keys() != ['fields']:
+            msg = (
+                '{0} provided as argument(s).  The \"special\" index only '
+                'contains the \"fields\" argument.'
+            ).format(kwargs)
+            raise CloudantArgumentError(msg)
+        if kwargs['fields'] != [{'_id': 'asc'}]:
```

Comment:
This is a pretty specific kwarg, since it is limited, I would maybe make this a boolean in the call like ascending_order=True and then if it is True, just set the fields content to {'_id': 'asc'} 
Make the API feel more like an API than exposing deeper knowledge of the special index, goal is to make it easy for the library user.